internal/impl/devbox.go

```diff
@@ -88,7 +89,17 @@ func Open(opts *devopt.Opts) (*Devbox, error) {
 		writer:        opts.Writer,
 		pure:          opts.Pure,
 	}
-	lock, err := lock.GetFile(box, searcher.Client())
+
+	// TODO savil: this is bad for perf, and so remove before enabling feature.
+	// this hack is to workaround an import cycle: lock -> nix -> lock
+	userSystem := ""
+	if featureflag.RemoveNixpkgs.Enabled() {
+		userSystem, err = nix.System()
+		if err != nil {
+			return nil, err
+		}
+	}
+	lock, err := lock.GetFile(box, searcher.Client(), userSystem)
```

Comment:
Use interfaces to get around dependency cycles. For example if you add `devbox.System()` and update the interface in `lock` pkg you are all good.

Reply:
Ugh, no, please lets not add more cruft to the `impl` package. 

Instead, would prefer simpifying `nix` package so it is not dependent on `lockfile`. From inspection, it the `profile` functions in `nix` have a lot of devbox-specific logic and so should be moved out into a different package.